sonar-scala-plugin/src/main/java/org/sonarsource/scala/externalreport/scalastyle/ScalastyleFamilySensor.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.scala.externalreport.scalastyle;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.xml.stream.XMLStreamException;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.batch.sensor.issue.NewExternalIssue;
+import org.sonar.api.batch.sensor.issue.NewIssueLocation;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonarsource.analyzer.commons.ExternalReportProvider;
+import org.sonarsource.analyzer.commons.ExternalRuleLoader;
+import org.sonarsource.scala.plugin.ScalaPlugin;
+
+public class ScalastyleFamilySensor implements Sensor {
+
+  private static final Logger LOG = Loggers.get(ScalastyleFamilySensor.class);
+
+  static final String SCALASTYLE_LINTER_KEY = "scalastyle";
+  static final String SCALASTYLE_LINTER_NAME = "Scalastyle";
+
+  static final String SCAPEGOAT_LINTER_KEY = "scapegoat";
+  static final String SCAPEGOAT_LINTER_NAME = "Scapegoat";
+
+  private static final String SCALASTYLE_PACKAGE_PREFIX = "org.scalastyle.";
+  private static final String SCAPEGOAT_PACKAGE_PREFIX = "com.sksamuel.scapegoat.inspections.";
+
+  public static final String REPORT_PROPERTY_KEY = "sonar.scala.scalastyle.reportPaths";
```

Comment:
It seems weird from a user perspective to specify the path of a scapegoat report to a property named "sonar.scala.scalastyle.reportPaths".